### ISSUE TITLE
Add collections:Map class

### ISSUE THREAD

**Issue by narqo** [bem/bem-core#870]
TL;DR 
To reduce some copy & paste within the library, we can add a new `collections:Map` class, with polyfill of the [native Map object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

Currently, we have several places within the codebase, which workaround the same behaviour, the native `Map`s provides, using the `identify` and simple Hash-object as a storage (see `events` and `i-bem*` modules for examples). It seems that we can incapsulate such things in our own `collection:Map` class, which would use native `Map`s if possible (see [browsers compatibility table](http://kangax.github.io/compat-table/es6/#Map)) and identify-based solution if not.

**Comment by veged** [comment]
close us discussed with @narqo and @dfilatov
